Class8_Sangavi

```diff
@@ -0,0 +1,71 @@
+-----------------------Number Check-------------------------------------
+import string
+import re
+
+def numcheck(str1):
+    if str == "":
+        return False
+    elif str1.count(".")>1 or str1.endswith("."):
+        return False
+    elif str1.count("-") and str1.count("+") == 1:
+        return False
+    elif str1.count("-") or str1.count("+") == 1:
+        if str1.startswith("-") or str1.startswith("+"):
+            return True
+    else:
+        return False
+    return True
+
+str1 = "+123001.29"
+print numcheck(str1)
+
+
+s = re.match(r'[+|-]?[0-9]+[\.]?[0-9]+$',str1)     #regex pattern
```

Comment:
Anna, the one above this regex is normal python program without regex.

RegEx:     re.match(r'[+|-]?[0-9]+[\.]?[0-9]+$',str1) 
re.match - returns the object if it finds the exact match of the given expression.

[+|-]? - the symbol "|" is to mention either "+" or "-".    
             "?" refers to optional. [+|-] may or may not be there.
[0-9] +   - can have any number between 0 and 9.
                "+" refers that the number(0-9) can occur one or more number of times. 
[\.]?  - "." is optional. may or may not be there
$-tells the end.
str1 - the input string to check

[+|-]?    [0-9]+     [\.]?     [0-9]+  
+           12           .            345      ----  +12.345
-             9            .            5          ----  -9.5
              1                        98        -----   198